homework_2/store/src/App.js

```diff
@@ -0,0 +1,137 @@
+import React, {Component} from 'react';
+
+import './App.css';
+
+import {Home} from './pages/home';
+import {User} from './pages/user';
+import {Admin} from './pages/admin';
+import {Cart} from './pages/cart';
+
+export class App extends Component {
+
+    products = [
+        {
+            name: 'MacBook pro 2018',
+            inCart: false
+        },
+        {
+            name: 'Dell xs',
+            inCart: false
+        },
+        {
+            name: 'Microsoft Surface',
+            inCart: false
+        }
+    ]
+
+    constructor() {
+        super();
+
+        this.state = {
+            activePage: 'homePage',
+            products:   this.products,
+            newProduct: '',
+            inCart: []
+        }
+    }
+
+    changePageToAdmin = () => {
+        this.setState({
+            activePage: 'adminPage'
+        });
+    }
+
+    changePageToUser = () => {
+        this.setState({
+            activePage: 'userPage'
+        });
+    }
+
+    changePageToCart = () => {
+        this.setState({
+            activePage: 'cartPage'
+        });
+    }
+
+    logOut = () => {
+        this.setState({
+            activePage: 'homePage'
+        });
+    }
+
+    addProduct = () => {
+        this.state.products.push({name: this.state.newProduct})
+
+        this.setState({
+            products: this.state.products,
+            newProduct: ''
+        })
+    }
+  
+    addProductName = (event) => {
+        this.setState({
+            newProduct: event.target.value
+        })
+    }
+
+    addToCart = (product) => {
+        let self = this;
+        let item = this.state.products.find(function(item, index){
+            if (item.name == product.name) {
+                self.state.products[index].inCart = true;
```

Comment:
if you have inCart in state and inCart in the object it seems like a duplication an additional overhead
Furthermore, it breaks one of principle "Single Source of Truth"